platform/pulp/platform/models/task.py

```diff
@@ -0,0 +1,231 @@
+        (ACCEPTED, 'Accepted'),
+        (RUNNING, 'Running'),
+        (SUSPENDED, 'Suspended'),
+        (FINISHED, 'Finished'),
+        (ERROR, 'Error'),
+        (CANCELED, 'Canceled')
+    )
+    type = models.TextField()
+    group = models.UUIDField()
+    state = models.TextField(choices=STATES)
+
+    started_at = models.DateTimeField()
+    finished_at = models.DateTimeField()
+
+    error = JSONField()
+    result = JSONField()
+
+    parent = models.ForeignKey("Task", null=True)
+    worker = models.ForeignKey("Worker")
+
+
+class TaskTag(Model):
+    """
+    Custom tags for a task
+
+    Fields:
+
+    :cvar name: The name of the tag
+    :type name: models.TextField
+
+    Relations:
+
+    :cvar task: The task this tag is associated with
+    :type task: models.ForeignKey
+    """
+    name = models.TextField()
+
+    task = models.ForeignKey(Task, related_name="tags", related_query_name="tag")
+
+
+class ScheduledCalls(Model):
+    """
+    Scheduled Call Request
+
+    Fields:
+
+    :cvar task: The task that should be run on a schedule
+    :type task: models.TextField
+
+    :cvar enabled: Indicates if schedule should be actively run by the scheduler
+    :type enabled: models.BooleanField
+
+    :cvar resource: Indicates a unique resource that should be used to find this schedule
+    :type resource: models.TextField
+
+    :cvar consecutive_failures: Number of times this task has failed consecutively
+    :type consecutive_failures: models.IntegerField
+
+    :cvar failure_threshold: Number of consecutive failures before task is automatically disabled
+    :type failure_threshold: models.IntegerField
+
+    :cvar iso_schedule: ISO8601 string representing the schedule
+    :type iso_schedule: models.TextField
+
+    :cvar schedule: Pickled instance of celery.schedules.schedule that should be run.
+    :type schedule: models.TextField
+
+    :cvar first_run: The first time this schedule was ran
+    :type first_run: models.DateTimeField
+
+    :cvar remaining_runs: Number of runs left until this schedule gets disabled
+    :type remaining_runs: models.IntegerField
+
+    :cvar total_run_count: Number of times this schedle has ran
+    :type total_run_count: models.IntegerField
+
+    :cvar last_run: Last time this schedule was ran
+    :type last_run: models.DateTimeField
+
+    :cvar last_updated: The last time this schedule was saved to the database
+    :type last_updated: models.DateTimeField
+
+    :cvar args: Arguments that should be passed to the apply_async function
+    :type args: models.JSONField
+
+    :cvar kwargs: Keyword arguments that should be passed to the apply_async function
+    :type kwargs: models.JSONField
+    """
+    task = models.TextField()
+    enabled = models.BooleanField(default=True)
+    resource = models.TextField(null=True)
+
+    consecutive_failures = models.IntegerField()
+    failure_threshold = models.IntegerField()
+
+    iso_schedule = models.TextField()
```

Comment:
This probably isn't a data layer concern except that I was hoping for more validation of this field.

For 3.0 we should contemplate if we want to continue to use ISO8601. I haven't met a person yet who can correctly form their intended schedule on the first try after reading the ISO8601 docs. Something like what cron uses would be so much better.

What do you think? Maybe we should send this to pulp-dev mailing list for comment?